WooCommerce/Classes/ViewRelated/Coupons/EnhancedCouponListViewController.swift

```diff
@@ -0,0 +1,187 @@
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        configureNavigation()
+        configureCouponListViewController()
+    }
+
+    /// Create a `UIBarButtonItem` to be used as the search button on the top-left.
+    ///
+    private lazy var searchBarButtonItem: UIBarButtonItem = {
+        let button = UIBarButtonItem(image: .searchBarButtonItemImage,
+                                     style: .plain,
+                                     target: self,
+                                     action: #selector(displaySearchCoupons))
+        button.accessibilityTraits = .button
+        button.accessibilityLabel = Localization.accessibilityLabelSearchCoupons
+        button.accessibilityHint = Localization.accessibilityHintSearchCoupons
+        button.accessibilityIdentifier = "coupon-search-button"
+
+        return button
+    }()
+
+    /// Create a `UIBarButtonItem` to be used as the create coupon button on the top-right.
+    ///
+    private lazy var createCouponButtonItem: UIBarButtonItem = {
+        let button = UIBarButtonItem(image: .plusImage,
+                style: .plain,
+                target: self,
+                action: #selector(displayCouponTypeBottomSheet))
+        button.accessibilityTraits = .button
+        button.accessibilityLabel = Localization.accessibilityLabelCreateCoupons
+        button.accessibilityHint = Localization.accessibilityHintCreateCoupons
+        button.accessibilityIdentifier = "coupon-create-button"
+
+        return button
+    }()
+}
+
+private extension EnhancedCouponListViewController {
+    func configureCouponListViewController() {
+        couponListViewController.view.translatesAutoresizingMaskIntoConstraints = false
+        addChild(couponListViewController)
+        view.addSubview(couponListViewController.view)
+        view.pinSubviewToAllEdges(couponListViewController.view)
+        couponListViewController.didMove(toParent: self)
+    }
+
+    func configureNavigation() {
+        title = Localization.title
+    }
+
+    func configureNavigationBarItems(hasCoupons: Bool) {
+        if hasCoupons {
+            navigationItem.rightBarButtonItems = [createCouponButtonItem, searchBarButtonItem]
+        } else {
+            navigationItem.rightBarButtonItems = [createCouponButtonItem]
+        }
+    }
+
+    func buildNoResultConfig() -> EmptyStateViewController.Config {
+        return .withButton(
+            message: .init(string: Localization.couponCreationSuggestionMessage),
+            image: .emptyCouponsImage,
+            details: Localization.emptyStateDetails,
+            buttonTitle: Localization.createCouponAction
+        ) { [weak self] button in
+            guard let self = self else { return }
+            self.displayCouponTypeBottomSheet()
+        }
+    }
+
+    /// Shows `SearchViewController`.
+    ///
+    @objc private func displaySearchCoupons() {
+        ServiceLocator.analytics.track(.couponsListSearchTapped)
```

Comment:
Perhaps not necessary since we only have 2 event track calls here, but we could inject Analytics in this class.

Reply:
Thanks for the comment. As you said with only two events I will leave it as it is, but I will have it into account in case we add more events to this class.